src/pages/blogs/how_shorebird_works.md

```diff
@@ -0,0 +1,229 @@
+
+Just-in-time systems have several nice properties. One is flexibility – a JIT’d
+language like JavaScript can run source code it’s never seen before in
+production. Another is that a JIT can be very good at "peak performance", since
+a JIT runtime includes a compiler (something to turn source code into machine
+code) during production, which means a sophisticated JIT can run some code,
+measure that it’s being run very often and then go back and compile the same
+code in a more optimized way (with different tradeoffs) to make it run faster (a
+type of "profile guided optimization"). In an "ahead of time" (AOT) compiled
+language (like Swift) the source code is only used on the developer’s machine to
+produce the "machine code" which will end up running on the user’s device. AOT
+solutions have the nice advantage of not including a compiler in production
+(smaller binary size) as well as having faster startup because there is no work
+to do when starting the app. At the tradeoff of some amount of peak performance
+as well as flexibility.
+
+Dart is an atypical language in that it has both JIT and AOT compiler workflows.
+Dart was designed originally as a JIT’d language, but is now most commonly used
+(as part of Flutter) with an AOT workflow. `flutter run –debug` uses Dart’s JIT
+mode, but `flutter build ipa` uses Dart’s AOT mode. Another side-effect of being
+a JIT is that typically much more information about the source code is kept
+around during production. This extra information is part of what enables a JIT
+to optimize hot functions, it’s also exactly the kind of information that has
+enabled what Shorebird has done.
+
+Functions within a JIT runtime need to be aware they could have different
+compiled representations (e.g. one simple compile and one later optimized
+compile for the same function). Shorebird takes advantage of this quirk of
+Dart’s architecture, to insert a new interpreter, as an alternative mechanism
+for a function to use to execute. This allows us to effectively replace parts of
+your application at runtime without needing to compile new code on the device.
+
+## An interpreter for Dart
+
+Adding an interpreter to Dart was a challenge (working on compilers is hard, if
+you like compilers, we’re [hiring](https://shorebird.dev/jobs/)), and took us
+most of the last year. We didn’t try to build a particularly fancy interpreter
+for Dart (that had been attempted at Google multiple times before), but rather
+built something very simple. One of the challenges this creates is that
+Interpreters (and particularly our simple one) are very slow. In our case, our
+current (unoptimized) interpreter is about 100x slower than executing Dart AOT
+code on a CPU. Thankfully we had an insight early on that made us not have to
+care about interpreter speed.
+
+The insight that makes Shorebird possible is building a system that runs only
+changed Dart logic on the interpreter. Since the vast majority of the
```

Comment:
```suggestion

This insight is that we can run only changed Dart logic on the interpreter, while continuing to run unchanged code on the CPU. Since the vast majority of the
```